worker/src/main/java/org/apache/celeborn/service/deploy/worker/memory/MemoryManager.java

```diff
@@ -93,6 +96,9 @@ public class MemoryManager {
   private long memoryFileStorageThreshold;
   private final LongAdder memoryFileStorageCounter = new LongAdder();
   private final StorageManager storageManager;
+  private boolean pinnedMemoryCheckEnabled;
+  private long pinnedMemoryCheckInterval;
```

Comment:
To avoid frequently calling a pinned memory counter,  I think you can cache the last pinned memory value and refresh it periodically. And exporting the pinned memory value to the metrics.

Reply:
Here is another PR introducing pinnedMemory metrics https://github.com/apache/celeborn/pull/3019

Reply:
getPinnedMemory is not called very frequently. It is called once every pinnedMemoryCheckInterval. The default is 10 seconds.